File: backend/genai_client.py
"""
Google Generative AI client initialization using settings configuration.
"""
from google import genai
from .config import settings
import os

# Set the environment variable for the API key
os.environ['GEMINI_API_KEY'] = settings.gemini_api_key

# Create the client instance using the new Google GenAI SDK
client = genai.Client()

File: backend/services/genaiClient.js
/**
 * Google Generative AI client initialization
 */
const { GoogleGenerativeAI } = require('@google/generative-ai');
const config = require('../config');

// Create the Generative AI client instance
const genAI = new GoogleGenerativeAI(config.geminiApiKey);

// Get the model instance
const model = genAI.getGenerativeModel({ model: 'gemini-2.5-flash' });

module.exports = {
    genAI,
    model
};
